executor/predictor/predictor_process.go

```diff
@@ -260,15 +260,30 @@ func (p *PredictorProcess) logPayload(nodeName string, logger *v1.Logger, reqTyp
 		Id:          guuid.New().String(),
 		SourceUri:   p.ServerUrl,
 		ModelId:     nodeName,
-		RequestId:   p.Ctx.Value(payload.SeldonPUIDHeader).(string),
+		RequestId:   puid,
 	})
 	return nil
 }
 
+func (p *PredictorProcess) getPUIDHeader() (string, error) {
+	// Check request ID is not nil
+	if puid, ok := p.Ctx.Value(payload.SeldonPUIDHeader).(string); ok {
```

Comment:
Will this not still panic if Value return nil? Should we not change for its existence before returning the string?

Reply:
Interestingly not...

There's a similar question on this [here](https://stackoverflow.com/questions/30162256/convert-nil-interface-to-pointer-of-something-in-golang).

>Using the "comma-ok" form:
The value of ok is true if the assertion holds. Otherwise it is false and the value of n is the zero value for type T. No run-time panic occurs in this case.